test/sender.test.js

```diff
@@ -42,7 +42,8 @@ describe('Sender', function () {
         write: (data) => {
           assert.strictEqual(data[0] & 0x40, 0x40);
           if (++count === 3) done();
-        }
+        },
+        on: () => {}
```

Comment:
To avoid adding this every time, we can use something like this

```js
class FakeSocket {
  constructor({ write, on } = {}) {
    if (write) this.write = write;
    if (on) this.on = on;
  }

  write() {}
  on() {}
}
```

and use it like

```js
const socket = new FakeSocket({
  write: (data) => {
    assert.strictEqual(data[0] & 0x40, 0x40);
    if (++count === 3) done();
  }
});
const sender = new Sender(socket, { 'permessage-deflate': perMessageDeflate });
```

Reply:
Sounds good, will switch them over.